xla/myplugin/BUILD

```diff
@@ -0,0 +1,35 @@
+load("//xla:xla.bzl", "xla_cc_binary", "xla_cc_test")
+
+cc_library(
+    name = "myplugin_cpp_pjrt",
+    srcs = [
+      "myplugin_cpp_pjrt.cc",
+    ],
+    hdrs = ["myplugin_cpp_pjrt.h"],
+    deps = [
+         "//xla/pjrt:pjrt_client",
+    ],
+)
+
+cc_library(
+    name = "myplugin_pjrt",
```

Comment:
How do we know this is a "working plugin"? Looks like no compile is implemented, is there any way to have a test that loads it, tries to pass a module, and gets unimplemented?

I guess that's uninteresting, but at least shows how to test a plugin.. Also, how do we test plugins?

Reply:
> How do we know this is a "working plugin"? Looks like no compile is implemented, is there any way to have a test that loads it, tries to pass a module, and gets unimplemented?

Depends on your definition of working. Indeed, you have to load JAX and import the module, which is listed in the bug on how to test. All I tested was, "it loaded and JAX returned unimplemented". The docs do explicitly state compile a JAX and load your module, which is what I did.

This PR is just the smallest possible PJRT implementation that can be loaded by JAX. Does that make sense? That's what I assumed was in my head was the request to dogfood the docs.

Reply:
Added some tests. 

> Also Is there any way to write a test in the same CL via bazel? I.e. (future work) should we make a "fake XLA PJRT framework" which serves a simple hard-coded blob of StableHLO and hardcoded input arguments? Or is PJRT only usable as a dynamically loaded thing?

That could work, the general advice at the moment is to build a shared lib so JAX can use it. Doesn't actually matter as that depends on distribution.